go/consensus/cometbft/apps/staking/state/state.go

```diff
@@ -1219,6 +1219,11 @@ func (s *MutableState) AddRewards(
 			continue
 		}
 
+		if q.Cmp(commonPool) == 1 {
```

Comment:
If this condition is true (`q > commonPool`), should we lower the reward and give as much as we can (`q = commmonPool.Clone()`)?

> // If an error occurs, the pool and affected accounts are left in an invalid state.
> // This may fail due to the common pool running out of stake.

The method's comment is not correct anymore, as the pool will never run out of stake.

This method is not fair. For example:
- If I have unlimited escrow balance, my rewards will always be too big and therefore skipped while others will receive their awards until they drain the pool.
- Assuming equal balances and too small pool to cover all rewards, the first in the list will get more rewards than the last one.

However, I'm not sure if this is a problem. If we want fairness, we should distribute the reminder of the pool according to stakes. But that is probably too much.

Reply:
Yeah, we should discuss what to do about this and implement it in a separate PR.  This PR just fixes the panic that would have happened otherwise :)